src/main/java/seedu/realodex/logic/parser/FilterCommandParser.java

```diff
@@ -130,6 +134,24 @@ private void checkValidNameIfApplicable(Prefix presentPrefix, List<String> keyph
         ParserUtil.parseName(name);
     }
 
+    /**
+     * Validates keyphrase if the present prefix is for a remarks.
+     * Each keyphrase must not be empty as per Remarks constraints.
+     *
+     * @param presentPrefix The prefix to check if it's tag-related.
+     * @param keyphrases The list of keyphrases representing potential tags.
+     * @throws ParseException if any tag keyphrase is invalid.
+     */
+    private void checkValidRemarkIfApplicable(Prefix presentPrefix, List<String> keyphrases) throws ParseException {
+        if (!presentPrefix.equals(PREFIX_REMARK)) {
+            return;
+        }
+        String keyphrase = keyphrases.get(keyphrases.size() - 1);
+        if (keyphrase.isEmpty()) {
+            throw new ParseException(MESSAGE_FILTER_EMPTY_REMARK);
+        }
+    }
+
```

Comment:
Good check for empty remark, do remember to reflect this on UG that empty remark is omitted on purpose as someone may interpret it as trying to list all clients instead

Reply:
Agreed!